What's new 2.20 (#1362)
run-ai/docs: docs/home/whats-new-2-20.md

## **Release Content** <date>

The Run:ai v2.20 What's New provides a detailed summary of the latest features, enhancements, and updates introduced in this version. They serve as a guide to help users, administrators, and researchers understand the new capabilities and how to leverage them for improved workload management, resource optimization, and more. 

!!! Important Note
    For a complete list of features that will be deprecated, see [Deprecation notifications](#deprecation-notifications).

### Researchers

#### Workloads - Workspaces and Training 

> *  **Stop/run actions for distributed workloads** - You can now stop and run distributed workloads from the UI and API. Scheduling rules for training workloads also apply to distributed workloads. This enhances control over distributed workloads, enabling greater flexibility and resource management. <span style="display:inline-block; background-color:white; color:#616161; padding:3px 8px; border-radius:3px; border:1px solid #616161; font-size:12px;">From cluster v2.20 onward</span>

Reviewer: ```suggestion
*  **Stop/run actions for distributed workloads** - You can now stop and run distributed workloads from the UI, CLI, and API. Scheduling rules for training workloads also apply to distributed workloads. This enhances control over distributed workloads, enabling greater flexibility and resource management. <span style="display:inline-block; background-color:white; color:#616161; padding:3px 8px; border-radius:3px; border:1px solid #616161; font-size:12px;">From cluster v2.20 onward</span>
```

* **Visibility into idle GPU devices** - Idle GPU devices are now displayed in in the UI and API showing the number of allocated GPU devices that have been idle for more than 5 minutes. This provides better visibility into resource utilization, enabling more efficient workload management.

* **Configurable workload completion with multiple runs** - You can now define the number of runs a training workload must complete to 
be considered finished directly in the UI, API, and CLI V2. Running training workloads multiple times improves 
the reliability and validity of training results. Additionally, you can configure how many runs can be scheduled in parallel, 
helping to significantly reduce training time and simplifying the process of managing jobs that require multiple runs.
See [standard training](../Researcher/workloads/training/standard-training/trainings-v2.md) for more details. <span style="display:inline-block; background-color:white; color:#616161; padding:3px 8px; border-radius:3px; border:1px solid #616161; font-size:12px;">From cluster v2.20 onward</span>

> * **Configurable grace period for workload preemption** - You can now set a grace period in the UI, API and CLI V2 providing a 

Reviewer: ```suggestion
* **Configurable grace period for workload preemption** - You can now set a grace period in the UI, API and CLIv2 providing a 
```

Reviewer: IMO should replace `CLI V2` with `CLIv2` in the doc

Author: @lavianalon it's up to you guys in relation to the product. I can change it to CLIv2 all over the docs to make sure it's consistent throughout.

Workload configurations combine swap settings with multi-GPU dynamic fractions, providing flexibility and efficiency for managing 
large-scale workloads. See [Multi-GPU memory swap](../Researcher/scheduling/gpu-memory-swap.md#multi-gpu-memory-swap). 
<span style="display:inline-block; background-color:white; color:#fc774a; padding:3px 8px; border-radius:3px; border:1px solid #fc774a; font-size:12px;">Beta</span> <span style="display:inline-block; background-color:white; color:#616161; padding:3px 8px; border-radius:3px; border:1px solid #616161; font-size:12px;">From cluster v2.20 onward</span>

#### Command Line Interface (CLI V2)

* **Support for Windows OS** - CLI V2 now supports Windows operating systems, enabling you to leverage the full 
capabilities of the CLI. <span style="display:inline-block; background-color:white; color:#616161; padding:3px 8px; border-radius:3px; border:1px solid #616161; font-size:12px;">From cluster v2.18 onward</span>

* **Unified training command structure** - Unified the `distributed` command into the `training` command to align with the Run:ai UI. 

> The `training` command now includes a new sub-command to support distributed workloads, ensuring a more consistent and streamlined user experience across both the CLI and UI.

Reviewer: Exists from v2.18, this is new compare to CLIv1

Author: I'll make it clear that it's for CLI v2

> * **New command for Kubernetes access** - Added a new CLI V2 command, `runai kubconfig set`, allowing users to set the 
> kubeconfig file with Run:ai authorization parameters. This enhancement enables users to gain access to the Kubernetes cluster, 
> simplifying authentication and integration with Run:ai-managed environments. 

Reviewer: ```suggestion
* **New command for Kubernetes access** - Added a new CLI V2 command, `runai kubconfig set`, allowing users to set the 
kubeconfig file with Run:ai authorization token. This enhancement enables users to gain access to the Kubernetes cluster, 
simplifying authentication and integration with Run:ai-managed environments. 
```

    * Metrics (over time)  
        * Project  
            * GPU_QUOTA 
            * CPU_QUOTA_MILLICORES
            * CPU_MEMORY_QUOTA_MB
            * GPU_ALLOCATION
            * CPU_ALLOCATION_MILLICORES
            * CPU_MEMORY_ALLOCATION_MB 
        * Department  
            * GPU_QUOTA 
            * CPU_QUOTA_MILLICORES
            * CPU_MEMORY_QUOTA_MB
            * GPU_ALLOCATION
            * CPU_ALLOCATION_MILLICORES
            * CPU_MEMORY_ALLOCATION_MB 

    * Telemetry (current time)  
        * Project  
            * GPU_QUOTA 
            * CPU_QUOTA 
            * MEMORY_QUOTA
            * GPU_ALLOCATION
            * CPU_ALLOCATION
            * MEMORY_ALLOCATION
            * GPU_ALLOCATION_NON_PREEMPTIBLE
            * CPU_ALLOCATION_NON_PREEMPTIBLE
            * MEMORY_ALLOCATION_NON_PREEMPTIBLE 
        * Department    
            * GPU_QUOTA 
            * CPU_QUOTA 
            * MEMORY_QUOTA
            * GPU_ALLOCATION
            * CPU_ALLOCATION
            * MEMORY_ALLOCATION
            * GPU_ALLOCATION_NON_PREEMPTIBLE
            * CPU_ALLOCATION_NON_PREEMPTIBLE
            * MEMORY_ALLOCATION_NON_PREEMPTIBLE 

## Deprecation notifications 

> ### Ongoing Dynamic MIG deprecation process

Reviewer: IMO, the deprecation section should be the first section in the doc

Author: @yuval-runai it's not really common practice to include deprecations before new features. I added a note at the beginning of the What's New saying they should take a look at it.
